From f6ec9e9b235cac7807f1f99ab3cd9647f766b091 Mon Sep 17 00:00:00 2001
From: lutong <11589481+luckye1@user.noreply.gitee.com>
Date: Thu, 15 May 2025 10:52:48 +0800
Subject: [PATCH 25/28] target/arm/kvm: Use PSCI_VERSIOn for version info

---
 target/arm/kvm.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/target/arm/kvm.c b/target/arm/kvm.c
index 0a66f91fcc..7be1a0f123 100644
--- a/target/arm/kvm.c
+++ b/target/arm/kvm.c
@@ -59,7 +59,6 @@
 #define ARM_SMCCC_KVM_FUNC_DISCOVER_IMPL_VER    64
 #define ARM_SMCCC_KVM_FUNC_DISCOVER_IMPL_CPUS   65
 
-#define ARM_SMCCC_KVM_DISCOVER_IMPL_VER_1_0     0x100000000
 
 #define ARM_SMCCC_CALL_VAL(type, calling_convention, owner, func_num) \
         (((type) << ARM_SMCCC_TYPE_SHIFT) | \
@@ -1108,7 +1107,7 @@ static void arm_handle_smcc_kvm_vendor_hypercall(ARMCPU *cpu)
                 return;
             }
             env->xregs[0] = SMCCC_RET_SUCCESS;
-            env->xregs[1] = ARM_SMCCC_KVM_DISCOVER_IMPL_VER_1_0;
+            env->xregs[1] = PSCI_VERSION(1, 0);
             env->xregs[2] = ms->target_ipml_cpu_num;
             break;
         case ARM_SMCCC_VENDOR_HYP_KVM_DISCOVER_IMPL_CPUS_FUNC_ID:
-- 
2.43.0

