From d6aaac41aa4da977a4c1b87e79d5147713dbfc3b Mon Sep 17 00:00:00 2001
From: lutong <11589481+luckye1@user.noreply.gitee.com>
Date: Thu, 15 May 2025 10:50:30 +0800
Subject: [PATCH 21/28] arm-qmp-cmds: introspection for ID register props

---
 target/arm/arm-qmp-cmds.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/target/arm/arm-qmp-cmds.c b/target/arm/arm-qmp-cmds.c
index b53d5efe13..abab5f6b42 100644
--- a/target/arm/arm-qmp-cmds.c
+++ b/target/arm/arm-qmp-cmds.c
@@ -21,6 +21,7 @@
  */
 
 #include "qemu/osdep.h"
+#include "qemu/error-report.h"
 #include "hw/boards.h"
 #include "kvm_arm.h"
 #include "qapi/error.h"
@@ -216,6 +217,23 @@ CpuModelExpansionInfo *qmp_query_cpu_model_expansion(CpuModelExpansionType type,
             qdict_put_obj(qdict_out, name, value);
         }
     }
+    /* If writable ID regs are supported, add them as well */
+    if (ARM_CPU(obj)->writable_id_regs == WRITABLE_ID_REGS_AVAIL) {
+        ObjectProperty *prop;
+        ObjectPropertyIterator iter;
+
+        object_property_iter_init(&iter, obj);
+
+        while ((prop = object_property_iter_next(&iter))) {
+            QObject *value;
+
+            if (!g_str_has_prefix(prop->name, "SYSREG_")) {
+                continue;
+            }
+            value = object_property_get_qobject(obj, prop->name, &error_abort);
+            qdict_put_obj(qdict_out, prop->name, value);
+        }
+    }
 
     if (!qdict_size(qdict_out)) {
         qobject_unref(qdict_out);
-- 
2.43.0

