From 2194fa07fadea1941ba258c380b64502b7a536f6 Mon Sep 17 00:00:00 2001
From: lixuyuan4 <lixuyuan4@huawei.com>
Date: Wed, 13 May 2026 17:15:54 +0800
Subject: [PATCH 07/11] libvirt: eliminate void* parameter in cpuTestCacheInfo

---
 tests/cputest.c | 25 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 8 deletions(-)

diff --git a/tests/cputest.c b/tests/cputest.c
index 92bdc61b89..e32fd14e50 100644
--- a/tests/cputest.c
+++ b/tests/cputest.c
@@ -373,9 +373,8 @@ cpuTestUpdate(const void *arg)
 
 
 static int
-cpuTestCacheInfo(const void *arg)
+cpuTestCacheInfo(const struct data *data)
 {
-    const struct data *data = arg;
     g_autoptr(virCPUDef) cpu = NULL;
     g_autofree char *result = NULL;
 
@@ -1109,12 +1108,22 @@ mymain(void)
     DO_TEST_UPDATE_ONLY(VIR_ARCH_X86_64, "host", "host-passthrough-features");
 
     /* cacheinfo parameter details tests */
-    DO_TEST(VIR_ARCH_X86_64, cpuTestCacheInfo,
-            "cacheinfo-full",
-            NULL, "cacheinfo-full", NULL, 0, NULL, 0, 0);
-    DO_TEST(VIR_ARCH_X86_64, cpuTestCacheInfo,
-            "cacheinfo-defaults",
-            NULL, "cacheinfo-defaults", NULL, 0, NULL, 0, 0);
+    do {
+        struct data data = { VIR_ARCH_X86_64, NULL, "cacheinfo-full", NULL, NULL, NULL, 0, 0, 0 };
+        g_autofree char *testLabel = NULL;
+        testLabel = g_strdup_printf("%s(%s): %s", "cpuTestCacheInfo",
+                                    virArchToString(VIR_ARCH_X86_64), "cacheinfo-full");
+        virTestRunLog(&ret, testLabel,
+                      (int (*)(const void *))cpuTestCacheInfo, &data);
+    } while (0);
+    do {
+        struct data data = { VIR_ARCH_X86_64, NULL, "cacheinfo-defaults", NULL, NULL, NULL, 0, 0, 0 };
+        g_autofree char *testLabel = NULL;
+        testLabel = g_strdup_printf("%s(%s): %s", "cpuTestCacheInfo",
+                                    virArchToString(VIR_ARCH_X86_64), "cacheinfo-defaults");
+        virTestRunLog(&ret, testLabel,
+                      (int (*)(const void *))cpuTestCacheInfo, &data);
+    } while (0);
 
     DO_TEST_UPDATE(VIR_ARCH_PPC64, "host", "guest", VIR_CPU_COMPARE_IDENTICAL);
     DO_TEST_UPDATE(VIR_ARCH_PPC64, "host", "guest-nofallback", VIR_CPU_COMPARE_INCOMPATIBLE);
-- 
2.34.1

